In [35]:
import pandas as pd
import os

In [36]:
from google.colab import drive 
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


Let's read one of the csv files we have.

In [37]:
df=pd.read_csv('gdrive/My Drive/collected_data/zara.csv', encoding= 'unicode_escape')

In [38]:
df.head()

Product       Price category
0                ROBE EN SATIN IMPRIMÉE  119,00 TND  dresses
1                          ROBE SATINÉE  119,00 TND  dresses
2             ROBE PORTEFEUILLE SATINÉE  119,00 TND  dresses
3  ROBE EN SATIN AVEC IMPRIMÉ À FLEURS  139,00 TND  dresses
4                          ROBE SATINÉE  139,00 TND  dresses

As shown in the dataframe, we have the name of the product, its price and its category.

In [39]:
df.tail()

Product      Price     category
185         CHOUCHOU FOULARD IMPRIMÉ  35,90 TND  accessories
186  TROUSSE MATELASSÉE ET IMPRIMÉE  35,90 TND  accessories
187  TROUSSE MATELASSÉE ET IMPRIMÉE  35,90 TND  accessories
188  TROUSSE MATELASSÉE ET IMPRIMÉE  59,90 TND  accessories
189  TROUSSE MATELASSÉE ET IMPRIMÉE  59,90 TND  accessories

In [40]:
df.shape

(190, 3)

In the first file which is corresponding to 'Zara', we have 190 elements.

Now, let's have a look at an other one.

In [41]:
df=pd.read_csv('gdrive/My Drive/collected_data/mango.csv', encoding= 'unicode_escape')

In [42]:
df.head()

Product    Price category
0  Robe texturée imprimée  119 TND  dresses
1  Robe texturée imprimée  119 TND  dresses
2   Salopette denim Wideleg  189 TND  dresses
3     Robe imprimée nœuds  139 TND  dresses
4     Robe imprimée nœuds  139 TND  dresses

In [43]:
df.tail()

Product      Price     category
395              Créoles irrégulières  45,90 TND  accessories
396              Créoles irrégulières  45,90 TND  accessories
397  Set boucles d’oreilles combinées  45,90 TND  accessories
398  Set boucles d’oreilles combinées  45,90 TND  accessories
399                             Créoles  45,90 TND  accessories

In [44]:
df.shape

(400, 3)

For mango, we have collected 400 elements.

Now, let's show all the files we have in our directory.

In [45]:
data = []

for filename in os.listdir('gdrive/My Drive/collected_data'):
    if filename.endswith("csv"): 
        # Your code comes here such as 
        print(filename)
        data.append(filename)
print(data)

zara.csv
shana.csv
mango.csv
HA.csv
drest.csv
['zara.csv', 'shana.csv', 'mango.csv', 'HA.csv', 'drest.csv']


In order to combine the 5 datasets into one, we need a reference for each one to know the source of the element. So, we are going to add the name of the file and the name of the retailer to each row. 

In [46]:
complete_dataframe=pd.DataFrame()
my_dataframes=[]
for filename in os.listdir('gdrive/My Drive/collected_data'):
    if filename.endswith("csv"): 
      path ='gdrive/My Drive/collected_data/'+filename
      df=pd.read_csv(path, encoding= 'unicode_escape')

      df['Filename'] = filename
      #if we remove '.csv' from the name of the file, we get the name of the retailer
      retailer_name=filename.replace(".csv","")
      df['Retailer']= retailer_name
      print(df.head(2))
      my_dataframes.append(df)

                   Product       Price category  Filename Retailer
0  ROBE EN SATIN IMPRIMÉE  119,00 TND  dresses  zara.csv     zara
1            ROBE SATINÉE  119,00 TND  dresses  zara.csv     zara
   Product       Price category   Filename Retailer
0  Legging  54,99 TND  dresses  shana.csv    shana
1      Top  24,99 TND  dresses  shana.csv    shana
                    Product    Price category   Filename Retailer
0  Robe texturée imprimée  119 TND  dresses  mango.csv    mango
1  Robe texturée imprimée  119 TND  dresses  mango.csv    mango
                           Product        Price category Filename Retailer
0       Pantacheville formel femme  69,990 TND    pants   HA.csv       HA
1  Jeans coupe High Rise Straight   69,990 TND    pants   HA.csv       HA
                                             Product  ... Retailer
0    Robe blouse noire en jean à manches bouffantes  ...    drest
1  NAFNAF – Robe d’été femme à manches tra...  ...    drest

[2 rows x 5 column

In [47]:
my_dataframes

[                                   Product       Price  ...  Filename Retailer
 0                  ROBE EN SATIN IMPRIMÉE  119,00 TND  ...  zara.csv     zara
 1                            ROBE SATINÉE  119,00 TND  ...  zara.csv     zara
 2               ROBE PORTEFEUILLE SATINÉE  119,00 TND  ...  zara.csv     zara
 3    ROBE EN SATIN AVEC IMPRIMÉ À FLEURS  139,00 TND  ...  zara.csv     zara
 4                            ROBE SATINÉE  139,00 TND  ...  zara.csv     zara
 ..                                     ...         ...  ...       ...      ...
 185              CHOUCHOU FOULARD IMPRIMÉ   35,90 TND  ...  zara.csv     zara
 186       TROUSSE MATELASSÉE ET IMPRIMÉE   35,90 TND  ...  zara.csv     zara
 187       TROUSSE MATELASSÉE ET IMPRIMÉE   35,90 TND  ...  zara.csv     zara
 188       TROUSSE MATELASSÉE ET IMPRIMÉE   59,90 TND  ...  zara.csv     zara
 189       TROUSSE MATELASSÉE ET IMPRIMÉE   59,90 TND  ...  zara.csv     zara
 
 [190 rows x 5 columns],
       Produc

In [48]:
#combining the 5 dataframes using the method concat from pandas 
complete_dataframe= pd.concat(my_dataframes)

In [49]:
complete_dataframe.head()

Product       Price  ...  Filename Retailer
0                ROBE EN SATIN IMPRIMÉE  119,00 TND  ...  zara.csv     zara
1                          ROBE SATINÉE  119,00 TND  ...  zara.csv     zara
2             ROBE PORTEFEUILLE SATINÉE  119,00 TND  ...  zara.csv     zara
3  ROBE EN SATIN AVEC IMPRIMÉ À FLEURS  139,00 TND  ...  zara.csv     zara
4                          ROBE SATINÉE  139,00 TND  ...  zara.csv     zara

[5 rows x 5 columns]

In [50]:
complete_dataframe.tail()

Product      Price  ...   Filename Retailer
357              GUILD – Doctor’s Bag White   32.9 TND  ...  drest.csv    drest
358  MOA – BOUCLE OREILLE COULEUR NOIR/MARRON   29.0 TND  ...  drest.csv    drest
359    MOA – BOUCLE OREILLE COULEUR OR/MARRON  119.9 TND  ...  drest.csv    drest
360    MOA – BOUCLE OREILLE COULEUR MARRON/OR   44.5 TND  ...  drest.csv    drest
361     MIGATO-Sac à main verni beige à rabat   31.2 TND  ...  drest.csv    drest

[5 rows x 5 columns]

In [51]:
complete_dataframe.shape

(1165, 5)

Our dataset have 1165 rows and 5 columns

In [52]:
complete_dataframe.to_csv('dataset.csv',index=False)